build/gen_stub.php

```diff
@@ -591,12 +829,43 @@ function parseFunctionLike(
                 }
                 $paramMeta[$varName]['preferRef'] = true;
             } else if ($tag->name === 'alias') {
-                $aliasParts = explode("::", $tag->getValue());
+
+                // @alias \Absolute\Another\function
+                // @alias \Absolute\Another\Classname::Method
+                // @alias Relative\function
+                // @alias Relative\Classname::Method
+                // @alias ReflectionClass::__clone
+                // @alias \ReflectionClass::__clone
+                // @alias functionName
+                // @alias \globalFunction
+                $aliasString = $tag->getValue();
+
+                // cut namespace
+                $namespaceParts = explode("\\", $aliasString);
+                $name = array_pop($namespaceParts);
+                // absolute
+                if ($namespaceParts && $namespaceParts[0] === "") {
+                    $namespace = array_slice($namespaceParts, 1);
+                } else {
+                    // relative
+                    if ($namespaceOrClassInfo instanceof ClassInfo) {
+                        $current_namespace = $namespaceOrClassInfo->getNamespace();
+                    } elseif ($namespaceOrClassInfo) {
+                        $current_namespace = $namespaceOrClassInfo;
+                    } else {
+                        $current_namespace = [];
+                    }
+
+                    $namespace = array_merge($current_namespace, $namespaceParts);
+                }
```

Comment:
This handling is incomplete, in that it will not respect `use` aliases. We should either define that `@alias` refers to a canonical (implicitly fully-qualified) name, or we should also take use aliases into account here. (https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/NameContext.php implements name resolution logic.)

Reply:
At the moment my focus was that the script is 100% compatible to the old one. And first of all it supports basic namespace.